pennylane/ops/op_math/linear_combination.py

```diff
@@ -0,0 +1,506 @@
+            Can be ``'qwc'`` (qubit-wise commuting), ``'commuting'``, or ``'anticommuting'``.
+        method (str): The graph coloring heuristic to use in solving minimum clique cover for grouping, which
+            can be ``'lf'`` (Largest First) or ``'rlf'`` (Recursive Largest First). Ignored if ``grouping_type=None``.
+        id (str): name to be assigned to this LinearCombination instance
+
+    **Example:**
+
+    A LinearCombination can be created by simply passing the list of coefficients
+    as well as the list of observables:
+
+    >>> coeffs = [0.2, -0.543]
+    >>> obs = [qml.X(0) @ qml.Z(1), qml.Z(0) @ qml.Hadamard(2)]
+    >>> H = qml.ops.LinearCombination(coeffs, obs)
+    >>> print(H)
+    0.2 * (X(0) @ Z(1)) + -0.543 * (Z(0) @ Hadamard(wires=[2]))
+
+
+    The coefficients can be a trainable tensor, for example:
+
+    >>> coeffs = tf.Variable([0.2, -0.543], dtype=tf.double)
+    >>> obs = [qml.X(0) @ qml.Z(1), qml.Z(0) @ qml.Hadamard(2)]
+    >>> H = qml.ops.LinearCombination(coeffs, obs)
+    >>> print(H)
+    0.2 * (X(0) @ Z(1)) + -0.543 * (Z(0) @ Hadamard(wires=[2]))
+
+
+    A LinearCombination can store information on which commuting observables should be measured together in
+    a circuit:
+
+    >>> obs = [qml.X(0), qml.X(1), qml.Z(0)]
+    >>> coeffs = np.array([1., 2., 3.])
+    >>> H = qml.ops.LinearCombination(coeffs, obs, grouping_type='qwc')
+    >>> H.grouping_indices
+    ((0, 1), (2,))
+
+    This attribute can be used to compute groups of coefficients and observables:
+
+    >>> grouped_coeffs = [coeffs[list(indices)] for indices in H.grouping_indices]
+    >>> grouped_obs = [[H.ops[i] for i in indices] for indices in H.grouping_indices]
+    >>> grouped_coeffs
+    [array([1., 2.]), array([3.])]
+    >>> grouped_obs
+    [[X(0), X(1)], [Z(0)]]
+
+    Devices that evaluate a LinearCombination expectation by splitting it into its local observables can
+    use this information to reduce the number of circuits evaluated.
+
+    Note that one can compute the ``grouping_indices`` for an already initialized LinearCombination by
+    using the :func:`compute_grouping <pennylane.LinearCombination.compute_grouping>` method.
+    """
+
+    num_wires = qml.operation.AnyWires
+    grad_method = "A"  # supports analytic gradients
+    batch_size = None
+    ndim_params = None  # could be (0,) * len(coeffs), but it is not needed. Define at class-level
+
+    def _flatten(self):
+        # note that we are unable to restore grouping type or method without creating new properties
+        return (self._coeffs, self._ops, self.data), (self.grouping_indices,)
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        new_op = cls(data[0], data[1])
+        new_op._grouping_indices = metadata[0]  # pylint: disable=protected-access
+        new_op.data = data[2]
+        return new_op
+
+    def __init__(
+        self,
+        coeffs,
+        observables: List[Operator],
+        simplify=False,
```

Comment:
Are we going to deprecate the `simplify` kwarg in favor of just `qml.simplify`?

Reply:
I would be happy with that, the simplification upon creation is a bit awkward right now. Though doing `return qml.expval(qml.simplify(qml.ops.LinearCombination(coeffs, ops)))` looks a bit awkward (as opposed to `return qml.expval(qml.ops.LinearCombination(coeffs, ops, simplify=True))` but this is more cosmetic and subjective

FYI I still have a problem for differentiability with just torch using simplify upon initialization currently. Only torch, likely due to going through the pauli rep.

Reply:
Yes, would be up for potentially deprecating that! I'll add it as a candidate for 0.37

Reply:
Should I add anything right away about deprecation?

Reply:
After discussion online decided not to do deprecations just yet